src/h3/_cy/error_system.pyx

```diff
@@ -0,0 +1,58 @@
+"""
+todo: use module docs to describe new error system
+
+- less aggressive correctness checking for indices
+- hierarchy?
+- error codes from C
+"""
+
+from .h3lib cimport H3Error
+
+class H3Exception(Exception):
+    pass
+
+class H3ValueError(H3Exception, ValueError):
+    pass
+
+class H3CellError(H3ValueError):
+    pass
+
+class H3EdgeError(H3ValueError):
+    pass
+
+class H3ResolutionError(H3ValueError):
+    pass
+
+class H3DistanceError(H3ValueError):
+    pass
+
+cdef check_for_error(H3Error err):
+    """
+    todo: more descriptive message
+    todo: add error codes as property
+    todo: allow passing in extra args for the error message?
+    """
+
+    d = {
```

Comment:
Does this require allocating the error objects and dictionary each time a check is made?

Reply:
Maybe. But there's also a chance Cython optimizes that away. I know they do that with if-else chains by converting them to case statements when possible. We should definitely change this later for performance if that isn't the case tho. At this point, I feel like I'm still just prototyping the idea of what the error interface looks like.

Reply:
Also improved on this by moving the mapping outside of the function. LMKWYT!